DuckDuckGo/AutomationServer.swift

```diff
@@ -0,0 +1,340 @@
+            Logger.automationServer.error("Failed to start listener: \(error)")
+            fatalError("Failed to start automation listener: \(error)")
+        }
+        listener.newConnectionHandler = handleConnection
+        listener.start(queue: .main)
+        // Output server started
+        Logger.automationServer.info("Automation server started on port \(port)")
+    }
+    
+    @MainActor
+    func receive(from connection: NWConnection) {
+        connection.receive(
+            minimumIncompleteLength: 1,
+            maximumLength: connection.maximumDatagramSize
+        ) { content, _, isComplete, error in
+            switch connection.state {
+            case .ready:
+                break // Connection is valid, continue
+            case .cancelled, .failed:
+                print("Connection is no longer valid \(connection.state) \(String(describing: error)) \(String(describing: content)).")
+                return
+            default:
+                print("Connection is in state \(connection.state).")
+                return
+            }
+            Logger.automationServer.info("Received request! \(String(describing: content)) \(isComplete) \(String(describing: error))")
+
+            if let error {
+                Logger.automationServer.error("Error: \(error)")
+                return
+            }
+
+            if let content {
+                Logger.automationServer.info("Handling content")
+                Task {
+                    await self.processContentWhenReady(connection: connection, content: content)
+                }
+            }
+
+            if !isComplete {
+                Logger.automationServer.info("Handling not complete")
+                self.receive(from: connection)
+            }
+        }
+    }
+    
+    @MainActor
+    func processContentWhenReady(connection: NWConnection, content: Data) async {
+        // Check if loading
+        while self.main.currentTab?.isLoading ?? false {
+            print("Still loading, waiting...")
+            try? await Task.sleep(nanoseconds: 200_000_000) // 0.2 seconds
+        }
+
+        // Proceed when loading is complete
+        Logger.automationServer.info("Handling content")
+        self.handleConnection(connection, content)
+    }
+    
+    func getQueryStringParameter(url: URLComponents, param: String) -> String? {
+        return url.queryItems?.first(where: { $0.name == param })?.value
+    }
+
+    @MainActor
+    func handleConnection(_ connection: NWConnection, _ content: Data) {
+        Logger.automationServer.info("Handling request!")
+        let stringContent = String(decoding: content, as: UTF8.self)
+        // Log first line of string:
+        if let firstLine = stringContent.components(separatedBy: CharacterSet.newlines).first {
+            Logger.automationServer.info("First line: \(firstLine)")
+        }
+
+        // Get url parameter from path
+        // GET / HTTP/1.1
+        if #available(iOS 16.0, *) {
```

Comment:
Rather than nested if elses... use a guard and return faster.

In this case
    guard #available(iOS 16.0, *) else { return }

This will remove 'pyramid of doom'